test/exercise/fp/solution.rb

```diff
@@ -4,12 +4,28 @@ class << self
       # Обратиться к параметрам фильма можно так:
       # film["name"], film["rating_kinopoisk"], film["rating_imdb"],
       # film["genres"], film["year"], film["access_level"], film["country"]
-      def rating(_array)
-        0
+      def rating(array)
+        array_raitings_with_nil = array.map do |film|
+          if film['country'].to_s.include?(',') && film['rating_kinopoisk'].to_f.positive?
+            film['rating_kinopoisk'].to_f
+          end
+        end
+
+        array_raitings = array_raitings_with_nil.select { |num| !num.nil? }
+
+        sum_rate = array_raitings.reduce(:+)
+
+        middle_rate = sum_rate / array_raitings.length
       end
 
-      def chars_count(_films, _threshold)
-        0
+      def chars_count(films, threshold)
+        array_films_names_with_nil = films.map { |film| film['name'].to_s if film['rating_kinopoisk'].to_f >= threshold }
```

Comment:
lets select them map

Reply:
@artempartos
Sorry, din't get it... Should I split up: 
`array_films_names_with_nil` = films.map {....`
into 
`array.select {...` 
and then 
`array.map`